src/mpid/ch4/shm/ipc/src/ipc_p2p.h

```diff
@@ -130,6 +130,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPCI_copy_data(MPIDI_IPC_hdr * ipc_hdr, MPIR_
                                         MPIDIG_REQUEST(rreq, datatype), 0, &actual_unpack_bytes,
                                         MPIR_TYPEREP_FLAG_NONE);
         MPIR_ERR_CHECK(mpi_errno);
+        if (actual_unpack_bytes < src_data_sz) {
+            MPIR_ERR_SETANDJUMP(mpi_errno, MPI_ERR_TRUNCATE, "**truncate");
+        }
```

Comment:
I think `**dtypemismatch` is probably better here. We don't know if the recv buffer is actually overflowed, only that the unpack didn't go as expected.

Reply:
Good point! Fixed.